### PR TITLE
Agency bug, tests, validation bugs

### PR DESCRIPTION
**Pull request by jdtoombs**

### CODE REVIEW

File: frontend/src/components/common/form/AutoCompleteText.tsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ export type IAutoCompleteProps = {
   required?: boolean;
```
Comment (jdtoombs) [inline frontend/src/components/common/form/AutoCompleteText.tsx:16]: This felt very hacky, this is for [this story](https://pimsteam.atlassian.net/browse/PIMS-2890?atlOrigin=eyJpIjoiMGNhZGYwYTAwMzFjNDRhY2IxOTM5MGZlYzU2YjA1ZDMiLCJwIjoiaiJ9). As agency and sub agency share the same `agencyId` I struggled a little bit to come up with the best solution. I originally tried filtering the agencies from the land form but we need to have all the agencies in order to set the parent agency etc.

Comment (Fosol) [reply frontend/src/components/common/form/AutoCompleteText.tsx:16]: The lookup has the `parentId`.  So you should always be able to filter.  If they select a sub-agency, then it should automatically set the **Agency** to the parent, and the **Sub-Agency** to the selected value.

Comment (jdtoombs) [reply frontend/src/components/common/form/AutoCompleteText.tsx:16]: Ok should be fine then, does this all currently. Just wasn't sure if I was adding too much logic to the autocomplete component

Comment (asanchezr) [reply frontend/src/components/common/form/AutoCompleteText.tsx:16]: It does feel like a lot of "custom logic" has been added to a "generic" auto-complete but I don't have a better solution. Other than pushing the custom logic to a new "AgencyAutoComplete" and that might be too much work. I say we leave as-is for now

**Pull request merged**